src/org/infinity/gui/SortableTable.java

```diff
@@ -119,23 +127,15 @@ public Component getTableCellRendererComponent(JTable table, Object value,
 
   private final class SortableTableModel implements TableModel, Comparator<TableItem>
   {
-    private final List<TableModelListener> listeners = new ArrayList<TableModelListener>();
-    private final List<TableItem> tableItems = new ArrayList<TableItem>();
-    private final List<Class<? extends Object>> columnClasses;
-    private final List<String> columnNames;
+    private final List<TableModelListener> listeners = new ArrayList<>();
+    private final List<TableItem> tableItems = new ArrayList<>();
+    private final Class<?>[] columnClasses;
+    private final String[] columnNames;
```

Comment:
The list of columns in all places constant therefore there is no sense to use dynamic lists